packages/monitors/src/kpi-monitor.ts

```diff
@@ -0,0 +1,330 @@
+import { JobOptions, MonitorJob, VolumeKPI } from './types';
+import GraphClient, {
+  Account,
+  DailyLendBorrowVolume,
+} from '@notional-finance/graph-client';
+import { log, submitMetrics } from '@notional-finance/logging';
+import { DDMetric } from '@notional-finance/logging';
+import { BigNumber, FixedNumber } from 'ethers';
+import { ExchangeRate } from '@notional-finance/durable-objects';
+
+const secondsInDay = 86400;
+
+const exchangeRatesUSD = new Map<string, ExchangeRate>();
+
+const series: DDMetric[] = [];
+
+const kpis = {
+  volume: {},
+  accounts: {},
+};
+
+async function processLoanVolumes(volumes: DailyLendBorrowVolume[]) {
+  // Group the volumes by period
+  const nowSeconds = new Date().getTime() / 1000;
+  const lastDay = nowSeconds - secondsInDay;
+  const lastWeek = nowSeconds - secondsInDay * 7;
+  const lastMonth = nowSeconds - secondsInDay * 30;
+
+  const volumeByPeriod = {
+    daily: volumes.filter((v) => v.date > lastDay),
+    weekly: volumes.filter((v) => v.date > lastWeek),
+    monthly: volumes.filter((v) => v.date > lastMonth),
+    total: volumes,
+  };
+
+  recordOverallVolume(volumes);
+  ['daily', 'weekly', 'monthly'].forEach((period) => {
+    recordVolumeByCurrency(volumeByPeriod[period], period);
+  });
+
+  console.log(`KPI: ${JSON.stringify(kpis)}`);
+}
+
+function recordOverallVolume(volumes: DailyLendBorrowVolume[]) {
+  const volumeTotal = getLocalVolumeKPI(volumes);
+  kpis.volume.overall = Object.keys(volumeTotal).reduce((acc, key) => {
+    if (key !== 'decimals') {
+      acc[key] = FixedNumber.fromValue(volumeTotal[key], 8).toUnsafeFloat();
+    }
+    return acc;
+  }, {});
+
+  Object.keys(volumeTotal).forEach((key) => {
+    if (key === 'decimals') return;
+    const overallSeries = {
+      metric: `kpi.volume.overall.${key}`,
+      tags: [`type:overall`, `chain:mainnet`, `volumeType:${key}}`],
+      points: [
+        {
+          timestamp: Math.round(new Date().getTime() / 1000),
```

Comment:
Use a single timestamp rather than fetching a new one everytime so we get a synchronized recording.